src/Integrations/Clearbit/SettingsClearbit.php

```diff
@@ -0,0 +1,456 @@
+<?php
+
+/**
+ * Clearbit Settings class.
+ *
+ * @package EightshiftForms\Integrations\Clearbit
+ */
+
+declare(strict_types=1);
+
+namespace EightshiftForms\Integrations\Clearbit;
+
+use EightshiftForms\Hooks\Filters;
+use EightshiftForms\Settings\SettingsHelper;
+use EightshiftForms\Hooks\Variables;
+use EightshiftFormsVendor\EightshiftLibs\Services\ServiceInterface;
+
+/**
+ * SettingsClearbit class.
+ */
+class SettingsClearbit implements SettingsClearbitDataInterface, ServiceInterface
+{
+	/**
+	 * Use general helper trait.
+	 */
+	use SettingsHelper;
+
+	/**
+	 * Filter settings sidebar key.
+	 */
+	public const FILTER_SETTINGS_SIDEBAR_NAME = 'es_forms_settings_sidebar_clearbit';
+
+	/**
+	 * Filter global settings key.
+	 */
+	public const FILTER_SETTINGS_GLOBAL_NAME = 'es_forms_settings_global_clearbit';
+
+	/**
+	 * Filter settings global is Valid key.
+	 */
+	public const FILTER_SETTINGS_GLOBAL_IS_VALID_NAME = 'es_forms_settings_global_is_valid_clearbit';
+
+	/**
+	 * Filter settings integration use key.
+	 */
+	public const FILTER_SETTINGS_IS_VALID_NAME = 'es_forms_settings_is_valid_clearbit';
+
+	/**
+	 * Settings key.
+	 */
+	public const SETTINGS_TYPE_KEY = 'clearbit';
+
+	/**
+	 * Clearbit Use key.
+	 */
+	public const SETTINGS_CLEARBIT_USE_KEY = 'clearbit-use';
+
+	/**
+	 * API Key.
+	 */
+	public const SETTINGS_CLEARBIT_API_KEY_KEY = 'clearbit-api-key';
+
+	/**
+	 * Clearbit available keys key.
+	 */
+	public const SETTINGS_CLEARBIT_AVAILABLE_KEYS_KEY = 'clearbit-available-keys';
+
+	/**
+	 * Instance variable for Clearbit data.
+	 *
+	 * @var ClearbitClientInterface
+	 */
+	protected $clearbitClient;
+
+	/**
+	 * Create a new instance.
+	 *
+	 * @param ClearbitClientInterface $clearbitClient Inject Clearbit which holds Clearbit connect data.
+	 */
+	public function __construct(ClearbitClientInterface $clearbitClient)
+	{
+		$this->clearbitClient = $clearbitClient;
+	}
+
+	/**
+	 * Register all the hooks
+	 *
+	 * @return void
+	 */
+	public function register(): void
+	{
+		\add_filter(self::FILTER_SETTINGS_SIDEBAR_NAME, [$this, 'getSettingsSidebar']);
+		\add_filter(self::FILTER_SETTINGS_GLOBAL_NAME, [$this, 'getSettingsGlobalData']);
+		\add_filter(self::FILTER_SETTINGS_IS_VALID_NAME, [$this, 'isSettingsValid'], 10, 2);
+		\add_filter(self::FILTER_SETTINGS_GLOBAL_IS_VALID_NAME, [$this, 'isSettingsGlobalValid']);
+	}
+
+	/**
+	 * Determine if settings are valid.
+	 *
+	 * @param string $formId Form ID.
+	 * @param string $type Integration type.
+	 *
+	 * @return boolean
+	 */
+	public function isSettingsValid(string $formId, string $type): bool
```

Comment:
```suggestion
	public function areSettingsValid(string $formId, string $type): bool
```

You are checking multiple settings, no? If that's the case, plural seems more fitting here.

Reply:
this is consistent like this across all integrations and settings so i would like to leave it as is

Reply:
Since the eightshift forms are still in their alpha phase, you can just rename all of those method names. You're not breaking anything by doing that.

Reply:
yes but we have on some places one setting that needs to be valid and on some multiple settings, but in the usage I don't care about that, I just care that it is valid. So I want to have a consistant method name for valid.

I will leave it as is